File: src/main/java/c6h2cl2/asm/LoadingPluginSolidXp.java
package c6h2cl2.asm;

import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin;

import java.util.Map;

/**
 * Created by C6H2Cl2 on 2016/09/06.
 */
@IFMLLoadingPlugin.TransformerExclusions({"net.minecraftforge","java","org"})
public class LoadingPluginSolidXp implements IFMLLoadingPlugin {
    public static final String ASM_PACKAGE = "c6h2cl2.asm";
    @Override
    public String[] getASMTransformerClass() {
        return new String[]{ASM_PACKAGE + ".TransformerSolidXp"};
    }

    @Override
    public String getModContainerClass() {
        return "c6h2cl2.mod.SolidXp.SolidXpCore";
    }

    @Override
    public String getSetupClass() {
        return null;
    }

    @Override
    public void injectData(Map<String, Object> data) {

    }

    @Override
    public String getAccessTransformerClass() {
        return ASM_PACKAGE + ".TransformerSolidXp";
    }
}

File: src/main/kotlin/c6h2cl2/asm/LoadingPluginSolidXp.kt
package c6h2cl2.asm

import net.minecraftforge.fml.relauncher.IFMLLoadingPlugin

@IFMLLoadingPlugin.TransformerExclusions("net.minecraftforge", "java", "org")
class LoadingPluginSolidXp : IFMLLoadingPlugin {
    override fun getASMTransformerClass(): Array<String> {
        return arrayOf(ASM_PACKAGE + ".TransformerSolidXp")
    }

    override fun getModContainerClass(): String {
        return "c6h2cl2.mod.SolidXp.SolidXpCore"
    }

    override fun getSetupClass(): String? {
        return null
    }

    override fun injectData(data: Map<String, Any>) {

    }

    override fun getAccessTransformerClass(): String {
        return ASM_PACKAGE + ".TransformerSolidXp"
    }

    companion object {
        val ASM_PACKAGE = "c6h2cl2.asm"
    }
}
